addons/io_scene_gltf2/blender/exp/gltf2_blender_gather_primitives.py

```diff
@@ -43,48 +44,69 @@ def gather_primitives(
     :return: a list of glTF2 primitives
     """
     primitives = []
+
+    blender_primitives = __gather_cache_primitives(blender_mesh_name,
+        vertex_groups, modifiers, export_settings)
+
+    for internal_primitive in blender_primitives:
+        material_idx = internal_primitive['material']
+        double_sided = False
+        material = None
+        if material_idx is not None:
+            blender_material = bpy.data.materials[material_names[material_idx]]
+            if bpy.app.version < (2, 80, 0):
+                double_sided = bpy.data.meshes[blender_mesh_name].show_double_sided
+            else:
+                double_sided = not blender_material.use_backface_culling
+            material = gltf2_blender_gather_materials.gather_material(blender_material,
+                                                                  double_sided,
+                                                                  export_settings)
+
+        primitive = gltf2_io.MeshPrimitive(
+            attributes=internal_primitive['attributes'],
+            extensions=None,
+            extras=None,
+            indices=internal_primitive['indices'],
+            material=material,
+            mode=None,
+            targets=internal_primitive['targets']
+        )
+        primitives.append(primitive)
+
+    return primitives
+
+@cached
+def __gather_cache_primitives(
+        blender_mesh_name: str,
+        vertex_groups: Optional[bpy.types.VertexGroups],
+        modifiers: Optional[bpy.types.ObjectModifiers],
+        export_settings
+) -> List[dict]:
+    """
+    Gather parts that are identical for instances, i.e. excluding materials
+    """
+    blender_mesh = bpy.data.meshes[blender_mesh_name]
+    primitives = []
+
     blender_primitives = gltf2_blender_extract.extract_primitives(
         None, blender_mesh, vertex_groups, modifiers, export_settings)
 
     for internal_primitive in blender_primitives:
-        if bpy.app.version < (2, 80, 0):
-            def __gather_materials_279(blender_primitive, blender_mesh, modifiers, export_settings):
-                if not blender_primitive['material']:
-                    # TODO: fix 'extract_primitives' so that the value of 'material' is None and not empty string
-                    return None
-                mesh_double_sided = blender_mesh.show_double_sided
```

Comment:
To be clear, we're dropping support for 2.79's mesh double-sided support here, right?  I'm OK with doing so, as the flag for it was removed from Blender 2.80.

Reply:
Oh it moved, sorry, didn't see that.